overlord/snapstate/backend/setup.go

```diff
@@ -111,6 +114,23 @@ func (b Backend) RemoveSnapFiles(s snap.PlaceInfo, typ snap.Type, meter progress
 	return nil
 }
 
+func (b Backend) RemoveSnapDir(s snap.PlaceInfo, otherInstances bool) error {
```

Comment:
I'm not a big fan of boolean arguments for functions. Let me show an example: `RemoveSnapDir(path, true)` is hard to parse unless the reader a) knows the function by heart (unlikely .) b) checks the function definition.  After reading the rest of the review I see this is never used "naked", always with a `otherInstances` var so its probably ok (even though I still don't like it that much :)

I also find the `otherInstances` not super clear. If true, does it mean "remove-snap-dir and all other instance dirs`? Or "remove-snap-dir but there are other instance dirs left". Would be great to find something that is more descriptive.

Reply:
Tweaked the naming a bit